src/blocks/vpn.rs

```diff
@@ -0,0 +1,300 @@
+//! ----|--------|--------
+//! `driver` | Which vpn should be used . Available drivers are: `"nordvpn"` | `"nordvpn"`
+//! `interval` | Update interval in seconds. | `10`
+//! `format_connected` | A string to customise the output in case the network is connected. See below for available placeholders. | `" VPN: $icon "`
+//! `format_disconnected` | A string to customise the output in case the network is disconnected. See below for available placeholders. | `" VPN: $icon "`
+//! `state_connected` | The widgets state if the vpn network is connected. | `info`
+//! `state_disconnected` | The widgets state if the vpn network is disconnected | `idle`
+//!
+//! Placeholder | Value                                                     | Type   | Unit
+//! ------------|-----------------------------------------------------------|--------|------
+//! `icon`      | A static icon                                             | Icon   | -
+//! `country`   | Country currently connected to                            | Text   | -
+//! `flag`      | Country specific flag (depends on a font supporting them) | Text   | -
+//!
+//! Action    | Default button | Description
+//! ----------|----------------|-----------------------------------
+//! `toggle`  | Left           | toggles the vpn network connection
+//!
+//! # Drivers
+//!
+//! ## nordvpn
+//! Behind the scenes the nordvpn driver uses the `nordvpn` command line binary. In order for this to work
+//! properly the binary should be executable without root privileges.
+//!
+//! # Example
+//!
+//! Shows the current vpn network state:
+//!
+//! ```toml
+//! [[block]]
+//! block = "vpn"
+//! driver = "nordvpn"
+//! interval = 10
+//! format_connected = "VPN: $icon "
+//! format_disconnected = "VPN: $icon "
+//! state_connected = "good"
+//! state_diconnected = "warning"
+//! ```
+//!
+//! Possible values for `state_connected` and `state_diconnected`:
+//!
+//! ```
+//! warning
+//! critical
+//! good
+//! info
+//! idle
+//! ```
+//! `Option::None` defaults to `idle`.
+//!
+//! # Icons Used
+//!
+//! - `net_vpn`
+//! - `net_wired`
+//! - `net_down`
+//! - country code flags (if supported by font)
+//!
+//! Flags: They are not icons but unicode glyphs. You will need a font that
+//! includes them. Tested with: <https://www.babelstone.co.uk/Fonts/Flags.html>
+
+use core::str::FromStr;
+use regex::Regex;
+use std::process::Stdio;
+use tokio::process::Command;
+
+use crate::util::country_flag_from_iso_code;
+
+use super::prelude::*;
+
+#[derive(Deserialize, Debug, SmartDefault)]
+#[serde(rename_all = "snake_case")]
+enum DriverType {
+    #[default]
+    Nordvpn,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(default)]
+pub struct Config {
+    driver: DriverType,
+    interval: Seconds,
+    format_connected: FormatConfig,
+    format_disconnected: FormatConfig,
+    state_connected: Option<String>,
+    state_disconnected: Option<String>,
+}
+
+impl Default for Config {
+    fn default() -> Config {
```

Comment:
Instead of manually implementing `Default` you can use `SmartDefault` macro. Example: https://github.com/greshake/i3status-rust/blob/c6a68b5d959019aebbda837db195bd5531b18b5a/src/blocks/cpu.rs#L52-L59

Reply:
I applied changes according to your suggestion.